src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -844,6 +844,12 @@ public override long GetBytes(int ordinal, long dataOffset, [CanBeNull] byte[] b
             var fieldDescription = RowDescription[ordinal];
             var handler = fieldDescription.Handler as ByteaHandler;
             if (handler == null)
+            {
```

Comment:
handler isn't actually used in this method - the cast to ByteaHandler is only used to check whether byte access should be permitted.

Let's replace this with a simple check that `fieldDescription.Handler` is either ByteaHandler or PostgisGeometryHandler.